common/templates/general.go

```diff
@@ -431,8 +431,41 @@ func CreateMessageSend(values ...interface{}) (*discordgo.MessageSend, error) {
 				}
 				msg.Components = append(msg.Components, discordgo.ActionsRow{[]discordgo.MessageComponent{menu}})
 			}
+		case "forward":
+			if val == nil {
+				continue
+			}
+			var m map[string]interface{}
+			switch t := val.(type) {
+			case SDict:
+				m = t
+			case *SDict:
+				m = *t
+			case map[string]interface{}:
+				m = t
+			default:
+				return nil, errors.New("invalid value passed to forward; must be an sdict with channel and message")
+			}
+
+			msg.Reference = &discordgo.MessageReference{
+				Type: 1,
+			}
+			for k, v := range m {
+				switch strings.ToLower(k) {
+				case "channel":
+					msg.Reference.ChannelID = ToInt64(v)
+					if msg.Reference.ChannelID <= 0 {
+						return nil, errors.New(fmt.Sprintf("invalid channel id '%s' provided to forward.", ToString(val)))
+					}
+				case "message":
+					msg.Reference.MessageID = ToInt64(v)
+					if msg.Reference.MessageID <= 0 {
+						return nil, errors.New(fmt.Sprintf("invalid message id '%s' provided to forward.", ToString(val)))
+					}
+				}
+			}
 		default:
-			return nil, errors.New(`invalid key "` + key + `" passed to send message builder`)
+			return nil, errors.New(`invalid key "` + key + `" passed to forward`)
```

Comment:
This error is for `complexMessage`, 'send message builder' is correct.

Reply:
Thanks for the catch, fixed.